pkg/render/applicationlayer/embed/embed.go

```diff
@@ -37,7 +37,7 @@ func init() {
 	}
 }
 
-func AsMap() (map[string]string, error) {
+func AsMap(fileSystem fs.FS) (map[string]string, error) {
```

Comment:
I think you can just add corerulesetFS as this package's global variable (like FS fs.FS above) so you don't need to have to specify this as a parameter.


You can also merge both FS'es together e.g.:

```golang
import (
	"embed"
	"fmt"
	"io/fs"
	
	"github.com/jcchavezs/mergefs"
	
	coreruleset "github.com/corazawaf/coraza-coreruleset/v4"
	
	corev1 "k8s.io/api/core/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

var (
	FS fs.FS

        coreruleset.FS
        
	//go:embed coreruleset
	crsFS embed.FS
)

func init() {
	crsSubFS, err := fs.Sub(crsFS, "coreruleset")
	if err != nil {
		panic(err)
 	}
 	
 	FS = mergefs.Merge(coreruleset.FS, crsSubFS)
 }
```

doing it this way you can make new file entry assertions in embed_test.go where you expect a bunch of files that should appear on in the FS object if you walk it

Reply:
im not sure I follow. The idea behind add the function parameter was so that I could generate a different configmap based on the coraza coreruleset as well as the existing configmap that contained `tigera.conf`. 
Would doing it this way not make it 1 configmap altogether?

Reply:
Got on a zoom with Lorcan, and he helped me undestand this better -- resolved